src/components/ConfigEditor/AzureCredentialsForm.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { ChangeEvent, FunctionComponent, useEffect, useMemo, useReducer, useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { InlineFormLabel, Button, Select, Input } from '@grafana/ui';
+
+import { AzureAuthType, AzureCredentials, isCredentialsComplete } from './AzureCredentials';
+
+export interface Props {
+  managedIdentityEnabled: boolean;
+  oboEnabled: boolean;
+  credentials: AzureCredentials;
+  azureCloudOptions?: SelectableValue[];
+  onCredentialsChange: (updatedCredentials: AzureCredentials) => void;
+  getSubscriptions?: () => Promise<SelectableValue[]>;
+}
+
+export const AzureCredentialsForm: FunctionComponent<Props> = (props: Props) => {
+  const { credentials, azureCloudOptions, onCredentialsChange, getSubscriptions } = props;
+  const hasRequiredFields = isCredentialsComplete(credentials);
+
+  const [subscriptions, setSubscriptions] = useState<Array<SelectableValue<string>>>([]);
+  const [loadSubscriptionsClicked, onLoadSubscriptions] = useReducer((val) => val + 1, 0);
+  useEffect(() => {
+    if (!getSubscriptions || !hasRequiredFields) {
+      updateSubscriptions([]);
+      return;
+    }
+    let canceled = false;
+    getSubscriptions().then((result) => {
+      if (!canceled) {
+        updateSubscriptions(result, loadSubscriptionsClicked);
+      }
+    });
+    return () => {
+      canceled = true;
+    };
+    // This effect is intended to be called only once initially and on Load Subscriptions click
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadSubscriptionsClicked]);
+
+  const authTypeOptions = useMemo<Array<SelectableValue<AzureAuthType>>>(() => {
+    let opts: Array<SelectableValue<AzureAuthType>> = [
+      {
+        value: 'clientsecret',
+        label: 'App Registration',
+      },
+    ];
+
+    if (props.managedIdentityEnabled) {
+      opts.unshift({
+        value: 'msi',
+        label: 'Managed Identity',
+      });
+    }
+
+    if (props.oboEnabled) {
+      opts.push({
+        value: 'clientsecret-obo',
+        label: 'App Registration (On-Behalf-Of)',
+      });
```

Comment:
I prefer the current UI of having the "on-behalf-of" as a toggle for two reasons:

 - It's easier to notice that it's a variant of the "App Registration" auth mode.
 - It's easier to add the "Alpha" warning (which is lost here).